AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Models/SQLModelValueConverterTests.swift

```diff
@@ -0,0 +1,54 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import XCTest
+
+@testable import Amplify
+@testable import AWSDataStoreCategoryPlugin
+
+class SQLModelValueConverterTests: XCTestCase {
+
+    let testDateString = "2020-02-02T08:00:00.000Z"
+
+    var testDate: Date {
+        return testDateString.iso8601Date!
+    }
+
+    func testModelWithEveryTypeConversionToBindings() {
+
+        // non-model type
+        let nonModelExample = ExampleNonModelType(someString: "string", someEnum: .foo)
+        let nonModelJSON = "{\"someString\":\"string\",\"someEnum\":\"foo\"}"
+
+        // example model
+        let example = ExampleWithEveryType(id: "df5dd4a4-34f8-4974-8a37-2617cf8dafe1",
```

Comment:
I went and provisioned the backend with NonModel type, it does not create any mutations or queries, and just defines the type. Seems like the usecase is for attaching it to an existing type that contains the model directive. for example:
```
type Todo @model {
  id: ID!
  name: String!
  description: String
  innerTodo: JustTodo
}

type JustTodo {
  name: String!
  description: Int
}
```
```
mutation createTodo($input: CreateTodoInput!) {
  createTodo(input: $input) {
    description
  	innerTodo {
      name
      description
    }
  }
}
```